### PR TITLE
[border-agent] use ephemeral port

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

File: tests/scripts/meshcop [hunk 1]
```diff
@@ -324,7 +324,7 @@ network_form()
     # verify mDNS is working as expected.
     local mdns_result="${TEST_BASE}"/mdns_result.log
     avahi-browse -aprt | tee "${mdns_result}"
-    grep -q "${OT_NETWORK_NAME}" "${mdns_result}"
+    OT_BORDER_AGENT_PORT=$(grep "${OT_NETWORK_NAME}" "${mdns_result}" | grep -o ';[0-9]\{5\};' | head -n1 | tr -d ';')
```
Comment (wgtdkp) [inline tests/scripts/meshcop:327]: nit: get the port with the `ba port` CLI command.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by librasungirl — APPROVED** [review]

File: tests/scripts/meshcop (hunk 1 above)
Comment (bukepo) [reply tests/scripts/meshcop:327]: This is intended behavior to verify the advertised port is correct.

**Review by jwhui — APPROVED** [review]

**Pull request merged**